### PR TITLE
[SIL-86] Add pull requests repository tests

### PR DESCRIPTION
**Pull request by kezc**
- created fake for `PullRequestDataSource`, but implemented only necessary methods :)
- added tests for `PullRequestsRepository`

### CODE REVIEW

File: app/src/test/java/com/appunite/loudius/fakes/FakePullRequestDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        )
+    }
+
+    private fun getPullRequestForId(id: Int) = PullRequest(
+        id = id,
+        draft = false,
+        number = id,
+        repositoryUrl = "https://api.github.com/repos/exampleOwner/exampleRepo${id}",
+        title = "example title $id",
+        ZonedDateTime.parse("2023-03-07T09:24:24Z").toLocalDateTime().plusHours(id.toLong()),
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/fakes/FakePullRequestDataSource.kt]: **Suggestion: ** You can make it shorter: 
`
        LocalDateTime.parse("2023-03-07T09:24:24").plusHours(id.toLong())
`
Note: There is no 'Z' at the end of string.

File: app/src/test/java/com/appunite/loudius/fakes/FakePullRequestDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    override suspend fun getPullRequestsForUser(author: String): Result<PullRequestsResponse> =
+        when (author) {
+            "exampleUser" -> getPullRequestResponse()
+            else -> Result.failure(WebException.NetworkError())
+        }
+
+    private fun getPullRequestResponse(): Result<PullRequestsResponse> {
+        val itemsAmount = 3
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/fakes/FakePullRequestDataSource.kt]: **Suggestion: ** You can move it to the function parameters with the default value as 3.

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private val userDataSource: UserDataSource = mockk {
+        coEvery { getUser() } returns Result.success(User(1, "exampleUser"))
+    }
+    private val repository = PullRequestsRepository(
+        pullRequestsNetworkDataSource = pullRequestsNetworkDataSource,
+        userDataSource = userDataSource
+    )
+
+    @Test
+    fun `GIVEN currently logged in user WHEN getting own pull requests THEN should return list of pull requests`() =
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: **Suggestion: ** Is the word `should` giving us more value after `THEN` here? 
In my opinion `currently` and `own` could be removed too.

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        runTest {
+            repository.getCurrentUserPullRequests()
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: **Suggestion:** I didn't think about it before. `getCurrentUserPullRequests()` could be renamed to `getPullRequests`. There wouldn't be any functionality for getting pull requests of another user.. at least for now.. 😅

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        }
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: **Issue:** Where is the test in that test? 😄

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    fun `GIVEN error during fetching user WHEN getting own pull requests THEN should return error`() =
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: **Suggestion:** `own` and `should` could be removed.  Should return error or failure?

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            val response = repository.getCurrentUserPullRequests()
+
+            Assertions.assertEquals(
+                Result.failure<PullRequestsResponse>(WebException.NetworkError()),
+                response
+            )
+        }
+
+    @Test
+    fun `GIVEN error during fetching pull requests WHEN getting own pull requests THEN should return list of pull requests`() =
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: **Suggestion**: `own` and `should` could be removed. The end of the test name is copied, should be something with an error. Gotcha. :P

**Review by Krzysiudan — COMMENTED** [review]
Someone was in a hurry :P

File: app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt (hunk 5 above)
Comment (kezc) [reply app/src/test/java/com/appunite/loudius/domain/PullRequestsRepositoryTest.kt]: got distracted probably . . .

**Review by Krzysiudan — APPROVED** [review]

**Comment by Krzysiudan** [comment]
Tests are failing 🤔

**Pull request merged**